docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,20 @@
+# Proof of work
```

Comment:
Why do we want to mix PoW with KYC PR? We have a separate PR for that https://github.com/FuelLabs/faucet/pull/48